src/subdevice/configuration.rs

```diff
@@ -14,7 +14,364 @@ use crate::{
     subdevice_state::SubDeviceState,
     sync_manager_channel::{Enable, Status, SyncManagerChannel, SM_BASE_ADDRESS, SM_TYPE_ADDRESS},
 };
-use core::ops::DerefMut;
+use core::{fmt::{Debug, Display}, ops::DerefMut};
+
+/// TODO
+pub trait ConfigureSubdevicePdi {
+    /// TODO
+    type Error;
+
+    /// TODO
+    async fn configure<'a, S>(
```

Comment:
Originally I would have had this just be an `FnMut` callback instead of its own trait but that's not possible at the moment, as far as I can figure out:
https://github.com/rust-lang/rust/issues/97362#issuecomment-2181747470